include/exadg/postprocessor/time_control.cpp

```diff
@@ -0,0 +1,130 @@
+/*  ______________________________________________________________________
+ *
+ *  ExaDG - High-Order Discontinuous Galerkin for the Exa-Scale
+ *
+ *  Copyright (C) 2021 by the ExaDG authors
+ *
+ *  This program is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ *  ______________________________________________________________________
+ */
+
+// C/C++
+#include <limits>
+
+// deal.II
+#include <deal.II/base/exceptions.h>
+
+// ExaDG
+#include <exadg/postprocessor/time_control.h>
+
+
+TimeControl::TimeControl()
+  : EPSILON(1.0e-10),
+    reset_counter(true),
+    calculate(false),
+    start_time(std::numeric_limits<double>::max()),
+    calculate_every_time_interval(-1.0),
+    calculate_every_time_steps(std::numeric_limits<unsigned int>::max()),
+    counter(0),
+    last_timestep_number_to_evaluate(dealii::numbers::invalid_unsigned_int)
+{
+}
+
+void
+TimeControl::reinit(bool const         calculate_,
+                    double const       start_time_,
+                    double const       calculate_every_time_interval_,
+                    unsigned int const calculate_every_time_steps_)
+{
+  calculate = calculate_;
+
+  if(calculate_)
```

Comment:
Question: I never used ExaDG for unsteady problems. Is this valid also in the unsteady case?

Reply:
Do you mean the boolean parameter or the check whether the parameters are consistent.

Reply:
Sorry something went wrong...